pkg/sysextutils/sysext_utils.go

```diff
@@ -38,23 +41,281 @@ func getID(name string) string {
+			}
+		}
+		return nil
+	})
+}
+
+// isStaticallyLinked determines if the specified binary is statically linked.
+func isStaticallyLinked(path string) bool {
+	f, err := elf.Open(path)
+	if err != nil {
+		logging.LogError("Failed to open file %s: %v", path, err)
+		return false // Assume not statically linked if unable to open
+	}
+	defer f.Close()
+
+	// Check if the INTERP section is present
+	section := f.Section(".interp")
+	if section == nil {
+		// No .interp section means it's likely a statically linked binary
+		return true
+	}
+
+	// Attempt to read data from the section to further verify
+	data, err := section.Data()
+	if err != nil {
+		logging.LogError("Failed to read .interp section data: %v", err)
+		return true // Assume statically linked if data cannot be read
+	}
+
+	// If the .interp section is empty, it's also considered statically linked
+	if len(data) == 0 {
+		return true
+	}
+
+	return false // .interp section is present and has data
+}
+
+func relocateAndPatchBinaries(rootfsDir, newRootPath string) error {
+	logging.Log("Starting to relocate and patch binaries in %s", rootfsDir)
+
+	err := filepath.Walk(rootfsDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			logging.LogError("Error accessing path %s: %v", path, err)
+			return err
+		}
+
+		if !info.IsDir() && isELFExecutable(path) {
+			if isStaticallyLinked(path) {
+				logging.Log("Skipping patching for statically linked binary: %s", path)
+				return nil
+			}
+
+			// Existing code for patching
+			logging.Log("Patching binary: %s", path)
+			if err := patchBinary(path, newRootPath); err != nil {
+				logging.LogError("Failed to patch binary %s: %v", path, err)
+				return err
+			}
+
+			if err := patchDependencies(path, newRootPath); err != nil {
+				logging.LogError("Failed to patch dependencies for %s: %v", path, err)
+				return err
+			}
+		}
+		return nil
+	})
+
+	if err != nil {
+		logging.LogError("Failed during the walking process in %s: %v", rootfsDir, err)
+		return err
+	}
+
+	logging.Log("Successfully relocated and patched all binaries in %s", rootfsDir)
+	return nil
+}
+
+// isELFExecutable checks if the file is an ELF executable
+func isELFExecutable(path string) bool {
+	f, err := os.Open(path)
+	if err != nil {
+		return false
+	}
+	defer f.Close()
+
+	elfFile, err := elf.NewFile(f)
+	if err != nil {
+		return false
+	}
+
+	return elfFile.Type == elf.ET_EXEC || elfFile.Type == elf.ET_DYN
+}
+
+// patchBinary patches the binary's interpreter and library paths
+func patchBinary(path, newRootPath string) error {
+	// Check if the binary is statically linked; if so, skip patching
+	if isStaticallyLinked(path) {
+		logging.Log("Skipping interpreter patching for statically linked binary: %s", path)
+		return nil
+	}
+
+	// Set the new interpreter path
+	// newInterpreterPath := filepath.Join(newRootPath, "lib64", "ld-linux-x86-64.so.2") // doesn't work
+	newInterpreterPath := "/lib64/ld-linux-x86-64.so.2" // works
```

Comment:
This seems like something that can change between systems, is there a way to make it configurable or add a way to automatically discover the interpreter path?

Reply:
If the goal is to use rewrite the interpreter path to be a prefixed version of the interpreter originally required by the binary, then `patchelf --print-interpreter` can be used to determine the original path ([here](https://github.com/flatcar/sysext-bakery/blob/e3534d36db3a2cde7604d52722e12ff52a4b7811/flix.sh#L113C14-L113C42) is an example of its usage for patching a binary to use dependencies from within a prefix).